src/lazy/expanded/macro_table.rs

```diff
@@ -204,504 +202,195 @@ impl MacroTable {
     pub const FIRST_USER_MACRO_ID: usize = Self::NUM_SYSTEM_MACROS;
 
     fn compile_system_macros() -> Vec<Arc<Macro>> {
-        // We cannot compile template macros from source (text Ion) because that would require a Reader,
-        // and a Reader would require system macros. To avoid this circular dependency, we manually
-        // compile any TemplateMacros ourselves.
-        vec![
+        let bootstrap_context = EncodingContext::empty();
+        let context = bootstrap_context.get_ref();
+
+        // Creates a `Macro` from a TDL expression
+        let template = |source: &str| {
+            Arc::new(Macro::from_template_macro(
+                TemplateCompiler::compile_from_source(context, source).unwrap(),
+            ))
+        };
+
+        // Creates a `Macro` whose implementation is provided by the system
+        let builtin = |name: &str,
+                       signature: &str,
+                       kind: MacroKind,
+                       expansion_analysis: ExpansionAnalysis| {
             Arc::new(Macro::named(
+                name,
+                TemplateCompiler::compile_signature(context, signature).unwrap(),
```

Comment:
🪧 Using the previously defined helper method, we can now compile the signature from source instead of manually instantiating a `Vec` of `Parameter`s.